sopel/plugins/rules.py

```diff
@@ -1158,36 +1166,63 @@ def is_global_rate_limited(self):
         rate_limit = datetime.timedelta(seconds=self._global_rate_limit)
         return self._metrics_global.is_limited(now - rate_limit)
 
+    def rate_limit_info(self, trigger: Trigger) -> RateLimitCheckInfo:
+        if trigger.admin or self.is_unblockable():
+            return RateLimitCheckInfo()
+
+        # TODO: these functions call now() and probably shouldn't
+        if self.is_user_rate_limited(trigger.nick):
+            template = self.get_user_rate_message(trigger.nick)
+        elif self.is_channel_rate_limited(trigger.sender):
+            template = self.get_channel_rate_message(trigger.nick, trigger.sender)
+        elif self.is_global_rate_limited():
+            template = self.get_global_rate_message(trigger.nick)
+        else:
+            return RateLimitCheckInfo()
+
+        # TODO: add:
+        #     * time_left
+        #     * time_left_sec
+        #     * rate_limit
+        #     * rate_limit_type
+        #     * command
+        #     * plugin
+        #     * ???
+        if template:
+            message = template.format(
+                nick=trigger.nick,
+                # TODO: add is_channel() to trigger?
```

Comment:
We haven't done this because it seems pretty redundant and pointless to have a method whose body can be implemented as `return !self.other_method()`.

Reply:
Fair enough, dropped the TODO, which I only committed by mistake anyway, it was there to remind me to ask about it :sweat_smile: